spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/channel/OracleChannelMessageStoreQueryProvider.java

```diff
@@ -53,15 +53,17 @@ public String getPollFromGroupQuery() {
 
 	@Override
 	public String getPriorityPollFromGroupExcludeIdsQuery() {
-		return "SELECT %PREFIX%CHANNEL_MESSAGE.MESSAGE_ID, %PREFIX%CHANNEL_MESSAGE.MESSAGE_BYTES from %PREFIX%CHANNEL_MESSAGE " +
+		return "SELECT /*+ INDEX(INT_CHANNEL_MESSAGE INT_CHANNEL_MSG_PRIORITY_IDX) */ " +
```

Comment:
Without the hint, it uses the PK which means a real sort is needed. If the DB uses the hint, it will still sort, but it's a no-op (linear traverse with no swaps) since the result set is already in the right order.

Feel free to reach out to your friend; we have a week before the release.